src/controllers/DynamicCategoryFacetQueryController.ts

```diff
@@ -0,0 +1,86 @@
+import { CategoryFacet } from '../ui/CategoryFacet/CategoryFacet';
+import { Assert } from '../misc/Assert';
+import { QueryBuilder } from '../ui/Base/QueryBuilder';
+import { IFacetRequest, IFacetRequestValue } from '../rest/Facet/FacetRequest';
+import { QueryEvents } from '../events/QueryEvents';
+import { FacetValueState } from '../rest/Facet/FacetValueState';
+
+// TODO: Create a real class and move in a separate file
+export type CategoryFacetValue = {
```

Comment:
this is just for reviewing purposes so you have an idea what the value is going to look like 😄

Reply:
Small point but could this be an interface starting with an `I`? Or is there a reason to use a `type` in this case? I saw it as a type of a function argument below, and I wasn't sure if it was a class or an interface at first glance.

Reply:
It's going to be a class this is just to show you what it contains it's the first thing removed in the next pr